tuning/utils/utils.py

```diff
@@ -46,3 +47,38 @@ def load_yaml_or_json(file_path: str) -> dict:
         if ext == ".json":
             return json.load(f)
     return None
+
+
+def validate_mergeable_datasets(datasets):
+    """Given list of datasets, validate if all datasets have same type and number of columns."""
+    if len(datasets) > 1:
+        ref_columns = datasets[0].features
+        ref_column_names = list(ref_columns.keys())
+        ref_column_types = {col: feat.dtype for col, feat in ref_columns.items()}
+
+        # Check all other datasets
+        for i, ds in enumerate(datasets[1:], start=2):
+            ds_column_names = list(ds.features.keys())
+            ds_column_types = {col: feat.dtype for col, feat in ds.features.items()}
+
+            # Check same set of columns
+            if set(ds_column_names) != set(ref_column_names):
+                logging.warning(
```

Comment:
Havn't we moved to using logger instead of logging directly?

Reply:
Yes, added it.